### PR TITLE
Payload loader

### LINKED ISSUE #7: [User Story] As a systems engineer, I would like to know the name and PID of the running processes on my remote device
### **Is your user story related to a problem? Please describe.**
As a systems engineer, I am frustrated by the steps needed to retrieve name and process ID (PID) of a running process on my remote rasberryPi.

### **Describe the solution you'd like**
I would like to see the names and PID's of running applications on my remote device in my database so that I can quickly refer and retrieve the information.

### **Describe alternatives you've considered**
Obtaining the information from a shell-only method is tedious and is not convenient.

### **Additional context**
The information can be sent to either SQL-style tables or as part of a MongoDB document.
Related to issue #1

### LINKED ISSUE #110: Add data received from daemon to DB
## **What needs to be completed?**
Server-side, interpret and insert the data payload delivered by daemon to the database.
Expand functionality found in `/server/api/daemon_endpoint.js`

Current payload format:
```json
{
   "processes": [
        {
            "name": "System Idle Process",
            "pid": 0
        },
    ...
    ]
}
```

#### **Related Issue(s):**
#7, #80

#### **Success Criteria**
- [x] Create schema for payload
- [x] Add mongoose logic to save payload to DB

### LINKED ISSUE #131: Add unique identifier for machine
## **What needs to be completed?**
Add a unique identifier for the daemon hosting machine. such that when the payload is delivered, items
Possible attributes include:
- motherboard serial no.
- cpu serial no.
- MAC address

#### **Related Issue(s):**
#110

#### **Success Criteria**
- [x] A machine unique attribute added to payload under `deviceId`

### PR DESCRIPTION
**Pull request by Pyxsys**
* Added unique machine identifier to daemon payload
* Added implementation to dameon_endpoint to add cpu logs to DB
  * Added helper function to cpu model to clarify responsibilities
* Added tests for new daemon feature
* Added tests for new api feature

closes #131 
closes #110
closes #7

### CODE REVIEW

File: server/models/process.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        type: String,
+      },
+      pid: {
+        type: Number,
+      },
+    },
+  ],
+})
+
+const Processes = mongoose.model('processes', ProcessSchema)
```
Comment (DerekRuiz) [inline server/models/process.js:16]: Does not require model if there is no collection for the Schema

File: server/models/cpu.js [hunk 2]
```diff
@@ -50,12 +51,41 @@ const CpuLogsSchema = new mongoose.Schema({
...
+    usageSpeed,
+    numProcesses,
+    threadsAlive,
+    threadsSleeping,
+    uptime,
+    timestamp,
+    processes: newProcesses,
+  })
+}
+
```
Comment (DerekRuiz) [inline server/models/cpu.js:84]: Functions do not belong in the models folder, relocate to API folder or a new file for extracting  information from the PI device
Model folder is only for defining schemas and models (collections)

**Review by DerekRuiz — CHANGES_REQUESTED** [review]
- [ ] Move the function to instantiate a new CpuLog object outside of the Model folder
- [x] If the Process schema does not require its own collection, remove the creating of `const Processes = mongoose.model('processes', ProcessSchema)`

The tests look proper

File: server/models/process.js (hunk 1 above)
Comment (Pyxsys) [reply server/models/process.js:16]: Process model was created as child for parent CpuLog.
A CpuLog contains many processes.

@DerekRuiz Please clarify.

Comment (Pyxsys) [reply server/models/process.js:16]: mongoose.model creates a doc in the DB. Will remove

File: server/models/cpu.js (hunk 2 above)
Comment (Pyxsys) [reply server/models/cpu.js:84]: After discussion,  extracted method will be moved to `api/daemon_endpoint`.

File: server/models/process.js (hunk 1 above)
Comment (Pyxsys) [reply server/models/process.js]: [Commit added](https://github.com/Pyxsys/ttg-healthcheck/pull/132/commits/ade3c48f53fa413188338190b2d3b04ca080e696)

File: server/models/cpu.js (hunk 2 above)
Comment (Pyxsys) [reply server/models/cpu.js]: [commit added](https://github.com/Pyxsys/ttg-healthcheck/pull/132/commits/ade3c48f53fa413188338190b2d3b04ca080e696)

**Review by DerekRuiz — APPROVED** [review]
Changes approved

**Review by shash3 — APPROVED** [review]
Everything looks good. I have 1 suggestion about the name of daemon_endpoint.js. I don't think we need to add "_endpoint" at the end as it is already in the api folder

**Pull request merged**